binary/src/Pos/Binary/Class/TH.hs

```diff
@@ -368,6 +368,12 @@ deriveSimpleBiInternal predsMB headTy constrs = do
                 when (length realFields /= length dcFields) $ templateHaskellError $
                     sformat ("Some field of "%shown
                     %" constructor doesn't have an explicit name") cName
+                when (length constrs > 1 && not (null realFields)) $
+                    templateHaskellError $
+                        sformat ("`deriveSimpleBi` no longer supports sum types \
+                                 \with named fields.\n Constructor "%shown%" has \
+                                 \named fields: "%shown%".")
+                        cName (map fst realFields)
```

Comment:
Maybe a mention of `deriveIndexedBi` in that error message so people don't have to go looking?

As simple as "Use derivedIndexedBi` instead. Probably a good idea to add a comment explaining `deriveIndexedBi` as well.

Reply:
Here's what the error message will look like
```
/Users/mhueschen/Projects/IOHK-work/cardano-sl/binary/test/Test/Pos/Binary/Cbor/CborSpec.hs:70:1: error:
    `deriveSimpleBi` no longer supports sum types with named fields.

Constructor Test.Pos.Binary.Cbor.CborSpec.Login has named fields: [Test.Pos.Binary.Cbor.CborSpec.login,Test.Pos.Binary.Cbor.CborSpec.age].

Please use `deriveIndexedBi` instead.
```